### ISSUE TITLE
Update Docker plugin to 4.5.1

### ISSUE THREAD

**Issue by nathan-knight** [pulumi/pulumi-awsx#1219]
### What happened?

I'm encountering the issue that is resolved by https://github.com/pulumi/pulumi-docker/pull/863 but I cannot use the latest plugin version that has this fix (4.5.1) because the AWSX plugin uses version 4.5.0 of the plugin regardless of which version of the `@pulumi/docker` package is installed. Can you please release a patch version that uses the newest version of the Docker plugin?

### Example

- Use `@pulumi/awsx` version 2.5.0 (the version of `@pulumi/docker` seems irrelevant to what plugin version gets used)
- `pulumi up`
- `pulumi plugin ls`, it will have used version `4.5.0` (I tried removing version 4.5.0 of the plugin and manually installing 4.5.1 but it would reinstall 4.5.0 and use it)

### Output of `pulumi about`

`pulumi about` doesn't work for me because I'm using a monorepo and my `yarn.lock` file is in a parent directory of the `Pulumi.yaml` file (perhaps it should search parent directories for the `yarn.lock` in addition to the `Pulumi.yaml` file). I was using `@pulumi/awsx` 2.5.0, `@pulumi/docker` 4.5.1, `pulumi` 3.105.0, among other things.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).
